### ISSUE TITLE
umqtt.simple: Subscribe and check_msg

### ISSUE THREAD

**Issue by adebaumann** [micropython/micropython-lib#256]
Good evening,

Maybe I'm suffering from not seeing the obvious, but how do I use subscriptions asynchronously? As far as I can see in the source code, the "subscribe"-routine ends in an endless "while 1:" loop with a "wait_msg()" inside.
This corresponds with my experience that once I call subscribe(), the rest of my script (except the callback function) is not executed. I've tried the examples too, removing everything after "subscribe(...)" does not seem to make any noticeable difference...

Can somebody help me out here? Maybe I've just been staring at the screen too long...

Cheers!

**Comment by zhujisheng** [comment]
you should use check_msg() instead of wait_msg(), if you want unblocked.

**Comment by adebaumann** [comment]
Yes, that much is clear. My problem is that "subscribe()" never actually returns because it ends with a "while 1:  op = self.wait_msg()..." block.

**Comment by EddieParis** [comment]
Your server seems to not reply to the subscription message sent 2 lines above.

Theoretically you have nothing more to do to subscribe to a topic. Check your server logs.
